File: 0234-palindrome-linked-list/0234-palindrome-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        
        ListNode* temp = head;
        int cnt = 0;
        while (temp) {
            temp = temp->next;
            cnt++;
        }

       
        temp = head;
        stack<int> st;
        int mid = cnt / 2;

        
        for (int i = 0; i < mid; i++) {
            st.push(temp->val);
            temp = temp->next;
        }

        
        if (cnt % 2 != 0) {
            temp = temp->next;
        }

        
        while (temp) {
            if (st.top() != temp->val) {
                return false;  
            }
            st.pop();
            temp = temp->next;
        }

        return true; 
    }
};

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean checkPalindrome(List<Integer> list){
        int i = 0;
        int j = list.size() - 1;
        while(i <= j){
            if(list.get(i) != list.get(j)) return false;
            i++;
            j--;
        }
        return true;
    }
    public boolean isPalindrome(ListNode head) {
        List<Integer> list = new ArrayList<>();
        ListNode temp = head;
        while(temp != null){
            list.add(temp.val);
            temp = temp.next;
        }
        boolean res = checkPalindrome(list);
        return res;
    }
}